godot-core/src/builtin/script.rs

```diff
@@ -0,0 +1,638 @@
+    pub(super) unsafe extern "C" fn set_func<T: ScriptInstance>(
+        p_instance: GDExtensionScriptInstanceDataPtr,
+        p_name: GDExtensionConstStringNamePtr,
+        p_value: GDExtensionConstVariantPtr,
+    ) -> GDExtensionBool {
+        let instance = instance_data_as_script_instance::<T>(p_instance);
+        let name = transfer_string_name_from_godot(p_name);
+        let value = &*Variant::ptr_from_sys(p_value);
+
+        let result = borrow_instance_mut(instance).set(name.clone(), value);
+
+        transfer_bool_to_godot(result)
+    }
+
+    /// # Safety
+    /// - the engine has to provide a valid pointer  fo the instance data and prop name
+    /// - a valid variant pointer for the return value is expected
+    pub(super) unsafe extern "C" fn get_func<T: ScriptInstance>(
+        p_instance: GDExtensionScriptInstanceDataPtr,
+        p_name: GDExtensionConstStringNamePtr,
+        r_ret: GDExtensionVariantPtr,
+    ) -> GDExtensionBool {
+        let instance = instance_data_as_script_instance::<T>(p_instance);
+        let name = transfer_string_name_from_godot(p_name);
+
+        let result = borrow_instance(instance)
+            .get(name.clone())
+            .map(|variant| transfer_variant_to_godot(variant, r_ret))
+            .is_some();
+
+        transfer_bool_to_godot(result)
+    }
+
+    /// # Safety
+    /// - the engine has to provide a valid pointer to the instance data
+    /// - a valid pointer for the returned item count is expected
+    pub(super) unsafe extern "C" fn get_property_list_func<T: ScriptInstance>(
+        p_instance: GDExtensionScriptInstanceDataPtr,
+        r_count: *mut u32,
+    ) -> *const GDExtensionPropertyInfo {
+        let instance = instance_data_as_script_instance::<T>(p_instance);
+
+        let property_list: Vec<_> = borrow_instance(instance)
+            .property_list()
+            .iter()
+            .map(|prop| prop.property_sys())
+            .collect();
+
+        transfer_ptr_list_to_godot(property_list, r_count)
+    }
+
+    /// # Safety
+    /// - the engine has to provide a valid pointer to the instance data
+    /// - a valid pointer for the returned item count is expected
+    pub(super) unsafe extern "C" fn get_method_list_func<T: ScriptInstance>(
+        p_instance: GDExtensionScriptInstanceDataPtr,
+        r_count: *mut u32,
+    ) -> *const GDExtensionMethodInfo {
+        let instance = instance_data_as_script_instance::<T>(p_instance);
+
+        let method_list: Vec<_> = borrow_instance(instance)
+            .method_list()
+            .iter()
+            .map(|method| method.method_sys())
+            .collect();
+
+        transfer_ptr_list_to_godot(method_list, r_count)
+    }
+
+    /// # Safety
+    /// - the engine has to provide a valid pointer to the instance data
+    /// - the length of the property info list is expect to not have changed since
+    ///   it was transfered to the engine.
```

Comment:
```suggestion
    /// # Safety
    /// - The engine has to provide a valid pointer to the instance data.
    /// - The length of the property info list is expected to not have changed since
    ///   it was transferred to the engine.
```